### ISSUE TITLE
Async function reported as "Special form" in HTML output.

### ISSUE THREAD

**Issue by jonathanslenders** [python/mypy#6765]
(I was asked to report the following as a bug.)

Take the following function:
```python
async def some_function() -> None:
    pass
```

It is reported as "Special form":
![image](https://user-images.githubusercontent.com/216638/57195630-2975cc80-6f55-11e9-9420-1e6cfc354c74.png)

This is something internal to mypy, and should not end up in the generated HTML.

**Comment by jonathanslenders** [comment]
Another similar issue:
Take this code:
```python
from typing import NamedTuple

def a() -> None:
    _FuzzyMatch(0, 0)

_FuzzyMatch = NamedTuple('_FuzzyMatch', [
    ('match_length', int),
    ('start_pos', int),
])

def b() -> None:
    _FuzzyMatch(0, 0)
```

![image](https://user-images.githubusercontent.com/216638/57200535-9c029e80-6f8d-11e9-8f38-fc15cf69bcae.png)

Somehow the forward reference becomes red...

**Comment by ivar-lazzaro** [comment]
looking

**Comment by ivar-lazzaro** [comment]
PR: https://github.com/python/mypy/pull/6777

**Comment by jonathanslenders** [comment]
Thanks!
